leshan-server-core/src/main/java/org/eclipse/leshan/server/registration/RegistrationUpdate.java

```diff
@@ -111,6 +122,17 @@ public BindingMode getBindingMode() {
         return objectLinks;
     }
 
+    public Map<String, String> getAdditionalAttributes() {
+        return additionalAttributes;
+    }
+
+    Map<String, String> updateAdditionalAttributes(Map<String, String> oldAdditionalAttributes) {
+
+        // putAll method updates already present key values or add them if not present.
```

Comment:
create a new map to keep immutability contract

Reply:
An immutable object is an object which can not be modified after its creation.

Generally, in Java it means all attributes of this object are final and "non modifiable".
In Java, you can create unmodifiable "view" of a  collection from a modifiable collection using `Collections.unmodifiablexxx`.

How do you do when you need to modify this kind of  object : you create a new one.
This is a bit less efficient about memory consumption/GC activity but this is more safe about thread and concurrent access.

So something like you propose is OK.
We could even imagine something a bit smarter in case of empty collection but this is probably premature optimization.

Reply:
Ok! Thank you for your explanation :-)